crates/services/shared-sequencer/src/service.rs

```diff
@@ -0,0 +1,266 @@
+//! Defines the logic how to interact with the shared sequencer.
+
+use crate::{
+    http_api::AccountMetadata,
+    ports::{
+        BlocksProvider,
+        Signer,
+    },
+    Client,
+    Config,
+};
+use async_trait::async_trait;
+use core::time::Duration;
+use fuel_core_services::{
+    stream::BoxStream,
+    EmptyShared,
+    RunnableService,
+    RunnableTask,
+    ServiceRunner,
+    StateWatcher,
+};
+use fuel_core_types::services::{
+    block_importer::SharedImportResult,
+    shared_sequencer::{
+        SSBlob,
+        SSBlobs,
+    },
+};
+use futures::StreamExt;
+use std::sync::Arc;
+
+/// Non-initialized shared sequencer task.
+pub struct NonInitializedTask<S> {
```

Comment:
Nit: I suggest renaming this to `UninitializedTask` to maintain parity with https://github.com/FuelLabs/fuel-core/pull/2442